File: regexlib/2021-5-15/javascript_test_file/regexlib_1803.js
/* 1803
 * [a-zA-Z0-9_\\-]+@([a-zA-Z0-9_\\-]+\\.)+(com)
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"a"*10000+"! _1SLQ_1"
 */
var REGEX = /[a-zA-Z0-9_\\-]+@([a-zA-Z0-9_\\-]+\\.)+(com)/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'a'.repeat(i*10000) + '! _1SLQ_1'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re_test_file/regexlib_1803.py
# 1803
# [a-zA-Z0-9_\\-]+@([a-zA-Z0-9_\\-]+\\.)+(com)
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:""+"a"*10000+"! _1SLQ_1"

import re
from time import perf_counter

regex = """[a-zA-Z0-9_\\-]+@([a-zA-Z0-9_\\-]+\\.)+(com)"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "a" * i * 10000 + "! _1SLQ_1"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")